### PR TITLE
VOLAPI-49 DELETE should delete

### PR DESCRIPTION
**Pull request by joshwilsdon**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
         "aperture-config": "git+https://github.com/joyent/aperture-config.git#master",
         "joyent-schemas": "git+https://github.com/joyent/schemas.git#caf3a226ed0707f5da897e1da151cc6d97fccda2",
         "jsprim": "0.6.1",
-        "verror": "1.6.1"
+        "verror": "1.10.0"
```
Comment (misterdjules) [inline package.json]: Just curious, why do we need a new version of verror? Is it to get the `hasCauseWithName` function?

File: test/volumes-basic.test.js [hunk 2]
```diff
@@ -275,30 +276,20 @@ if (CONFIG.experimental_nfs_shared_volumes !== true) {
...
-
-                            t.end();
-                    });
-            });
+    test('volume should eventually disappear', function (t) {
+        mod_testVolumes.waitForDeletion(CLIENT, testVolume.id,
+            function onDeleted() {
+                CLIENT.get('/my/volumes/' + testVolume.id,
+                    function onGetVolume(getVolumeErr) {
+                        t.ok(verror.hasCauseWithName(getVolumeErr,
```
Comment (misterdjules) [inline test/volumes-basic.test.js]: Doesn't `waitForDeletion` calls its callback with no error if the GetVolume responded with a `VOLUME_NOT_FOUNDError`?

File: package.json (hunk 1 above)
Comment (joshwilsdon) [inline package.json]: yep, it's for hasCauseWithName which came in 1.9.0. I figured might as well go with latest.

File: test/volumes-basic.test.js (hunk 2 above)
Comment (joshwilsdon) [inline test/volumes-basic.test.js]: Yes, but if you notice here onDeleted is the callback which was passed no arguments. Subsequently, we're doing a *CLIENT.get* and then checking that the GET also returned VOLUME_NOT_FOUNDError.

File: package.json (hunk 1 above)
Comment (misterdjules) [inline package.json]: Sounds good!

File: test/volumes-basic.test.js (hunk 2 above)
Comment (misterdjules) [inline test/volumes-basic.test.js]: Sorry, somehow I got confused. Now it looks good to me.

**Comment by joshwilsdon** [comment]
merged with 93155b4179c4c09118aae1d81c76b4b6adc899c2

**Pull request closed without merging**